File: app/src/main/java/ru/noties/adapt/sample/core/item/SectionItem.java
package ru.noties.adapt.sample.core.item;

import android.support.annotation.NonNull;

public class SectionItem extends Item {

    private final String name;

    public SectionItem(long id, @NonNull String name) {
        super(id);
        this.name = name;
    }

    @NonNull
    public String name() {
        return name;
    }
}

File: sample/src/main/java/io/noties/adapt/sample/items/SectionItem.kt
package io.noties.adapt.sample.items

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import io.noties.adapt.Item
import io.noties.adapt.sample.R
import java.util.*

class SectionItem(val text: String) :
    Item<SectionItem.Holder>(Objects.hash(SectionItem::class, text).toLong()) {

    class Holder(view: View) : Item.Holder(view) {
        val textView: TextView = requireView(R.id.text)
    }

    override fun createHolder(inflater: LayoutInflater, parent: ViewGroup): Holder {
        return Holder(inflater.inflate(R.layout.item_section, parent, false))
    }

    override fun render(holder: Holder) {
        holder.textView.text = text
    }
}